pymongo/mongo_client.py

```diff
@@ -738,7 +738,13 @@ def target():
             client = self_ref()
             if client is None:
                 return False  # Stop the executor.
-            MongoClient._process_periodic_tasks(client)
+            try:
+                MongoClient._process_periodic_tasks(client)
+            except InvalidOperation as e:
+                if client._get_topology._closed:
+                    return False
+                else:
+                    raise e
```

Comment:
I don't think this will actually work. The problem is that `_process_periodic_tasks` already catches and logs the InvalidOperation error which is want we want to prevent. We need to refactor this kind of logic so that we can ignore InvalidOperation without logging it at all:
```python
            try:
                self._cleanup_cursor(True, cursor_id, address, sock_mgr,
                                     None, False)
            except Exception:
                helpers._handle_exception()
```

It would also be good to add a test for this. Perhaps something like this would work but it requires some experimentation:
```python
coll.insert_many([{} for _ in range(5])
cursor = coll.find(batchSize=2)
cursor.next()
assert cursor.cursor_id
client.close()
# Add cursor to kill cursors queue 
del cursor
wait_until(cursor_id is added to client's kill cursors queue)
client._process_periodic_tasks()   # This must not raise or print any exceptions
```